File: src/main/java/aoc2019/day09/Output.java
package aoc2019.day09;

import lombok.Data;

@Data
public class Output {
    private String code;
    private int index;
    private int relativeBase;

    public Output(String code) {
        this.code = code;
    }

    public Output(int index) {
        this.index = index;
    }

    public Output(String code, int index) {
        this.code = code;
        this.index = index;
    }

    public Output(String code, long index) {
        this.code = code;
        this.index = (int) index;
    }

    public Output(String code, long index, int relativeBase) {
        this.code = code;
        this.index = (int) index;
        this.relativeBase = relativeBase;
    }

    @Override
    public String toString() {
        return "{" +
                "code='" + code + '\'' +
                ", index=" + index +
                ", relativeBase=" + relativeBase +
                '}';
    }
}

File: 2015-2023/src/main/kotlin/aoc2019/day09/Output.kt
package aoc2019.day09

import lombok.Data

@Data
class Output {
    var code: String? = null
    var index = 0
    var relativeBase = 0

    constructor(code: String?) {
        this.code = code
    }

    constructor(index: Int) {
        this.index = index
    }

    constructor(code: String?, index: Int) {
        this.code = code
        this.index = index
    }

    constructor(code: String?, index: Long) {
        this.code = code
        this.index = index.toInt()
    }

    constructor(code: String?, index: Long, relativeBase: Int) {
        this.code = code
        this.index = index.toInt()
        this.relativeBase = relativeBase
    }

    override fun toString(): String {
        return "{" +
                "code='" + code + '\'' +
                ", index=" + index +
                ", relativeBase=" + relativeBase +
                '}'
    }
}